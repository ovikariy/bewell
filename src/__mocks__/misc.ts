/*
  special mock for react-native-get-random-values otherwise gives:
  TypeError: Cannot read property 'getRandomBase64' of undefined
  at Object.<anonymous> (node_modules/react-native-get-random-values/getRandomBase64.js:1:18)
 */
jest.mock('react-native-get-random-values', () => ({
  getRandomBase64: jest.fn(),
}));

/* always keep date now the same because it would change in the test each time it runs and fail */
Date.now = jest.fn(() => 1482363367071);

export const StableDate = new Date(Date.now());