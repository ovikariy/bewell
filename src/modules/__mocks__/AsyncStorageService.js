const testData = {
    // 'key1': 'value1',
    // 'key2': 'value2'
};

export default function mockAsyncStorageService() {
    return {
        getItem: jest.fn((key) => { return testData[key]; }),
        multiGet: jest.fn((keys) => {
            const result = [];
            for(var i=0; i < keys.length; i++) {
                result.push([keys[i], testData[keys[i]]]);
            }
            return result;
        }),
        multiSet: jest.fn((items) => {
            items.forEach((item) => {
                testData[item[0]] = item[1];
            });
        }),
        multiRemove: jest.fn((keys) => {
            keys.forEach((key) => {
                delete testData[key];
            })
        }),
        getAllKeys: jest.fn(() => { return Object.keys(testData); })
    };
}