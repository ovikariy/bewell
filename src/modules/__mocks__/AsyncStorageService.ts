const data: { [key: string]: string } = {};

export function createMockAsyncStorageService() {
    return {
        getItem: jest.fn((key: string) => data[key]),
        multiGet: jest.fn((keys: string[]) => keys.map(key => data[key])),
        multiSet: jest.fn((items: [string, string][]) => {
            items.forEach(item => {
                data[item[0]] = item[1];
            });
        }),
        multiRemove: jest.fn((keys: string[]) => {
            keys.forEach(key => {
                delete data[key];
            });
        }),
        getAllKeys: jest.fn(() => Object.keys(data))
    };
}