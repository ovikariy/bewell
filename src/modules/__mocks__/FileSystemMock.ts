import { isNullOrEmpty } from "../utils";

interface FSInterface {
  freeDiskSpace: number,
  files: { [key: string]: string | null },
  directories: { [key: string]: string | null }
}

export const fs: FSInterface = {
  freeDiskSpace: 999999999999999,
  files: {
    file1: '{ key1: \'value1\' }',
    file2: '{ key2: \'value2\' }',
  },
  directories: {
    cacheDirectory: 'cache/',
    documentDirectory: 'documents/'
  }
};

export function createFileSystemMock() {
  return {
    EncodingType: {
      UTF8: "utf8",
      Base64: "base64"
    },
    WritingOptions: {
      encoding: {
        UTF8: "utf8",
        Base64: "base64"
      }
    },
    cacheDirectory: fs.directories.cacheDirectory,
    documentDirectory: fs.directories.documentDirectory,
    readAsStringAsync: jest.fn((fileUri: string, options?: any) => fs.files[fileUri] ? fs.files[fileUri] + '' : undefined),
    getFreeDiskStorageAsync: jest.fn(() => fs.freeDiskSpace),
    writeAsStringAsync: jest.fn((fileUri: string, contents: string, options?: any) => {
      fs.files[fileUri] = contents;
    }),
    getInfoAsync: jest.fn((fileUri: string, options?: { md5?: boolean, size?: boolean }) => {
      return {
        exists: (fs.files[fileUri] !== undefined || fs.directories[fileUri] !== undefined),
        size: (fs.files[fileUri] !== undefined || fs.directories[fileUri] !== undefined) ? 100 : 0,
        isDirectory: (fs.directories[fileUri] !== undefined) ? true : false,
        uri: fileUri
      };
    }),
    deleteAsync: jest.fn((fileUri: string) => {
      if (fs.files[fileUri])
        delete fs.files[fileUri];
    }),
    makeDirectoryAsync: jest.fn((fileUri: string, options?: { intermediates?: boolean }) => {
      fs.directories[fileUri] = fileUri;
    }),
    readDirectoryAsync: jest.fn((directoryUri: string) => {
      if (isNullOrEmpty(directoryUri))
        return [];
      const result = [];
      if (!directoryUri.endsWith('/') && !directoryUri.endsWith('\\'))
        directoryUri = directoryUri + '/';
      for (const key in fs.files) {
        if (key.startsWith(directoryUri))
          result.push(key.replace(directoryUri, '')); //we want only file names not full paths
      }
      return result;
    }),
    copyAsync: jest.fn((options: { from: string, to: string }) => {
      fs.files[options.to] = fs.files[options.from];
    })
  };
}