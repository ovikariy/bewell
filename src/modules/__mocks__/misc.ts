/*
  special mock for react-native-get-random-values otherwise gives:
  TypeError: Cannot read property 'getRandomBase64' of undefined
  at Object.<anonymous> (node_modules/react-native-get-random-values/getRandomBase64.js:1:18)
 */
jest.mock('react-native-get-random-values', () => ({
  getRandomBase64: jest.fn(),
}));