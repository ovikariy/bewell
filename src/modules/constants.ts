import { ErrorMessage } from "./translations";
export { ErrorMessage };

export const images = {
  logo_light: require('../assets/images/logo-light.png'),
  logo_dark: require('../assets/images/logo-dark.png'),
  logo_small_light: require('../assets/images/logo-small-light.png'),
  logo_small_dark: require('../assets/images/logo-small-dark.png'),
  splash: require('../assets/images/splash.png'),
  icon: require('../assets/images/icon.png'),
  header: getRandomHeaderImage(), /* chooses a random image from the folder for fun */
  arrow_up_dark: require('../assets/images/arrow-up-dark.png'),
  arrow_up_light: require('../assets/images/arrow-up-light.png'),
};

/* ItemTypes correspond to data keys in storage and will be hashed with DATA ENCRYPTION KEY
and their values will be encrypted with the DATA ENCRYPTION KEY;
These are only for widgets and not to be used for other storage items */
export const ItemTypes = {
  MOOD: 'MOOD',
  NOTE: 'NOTE',
  SLEEP: 'SLEEP',
  IMAGE: 'IMAGE'
};

export const brokenImageURI = 'broken'; /* this is just so that we can show something to user to indicate the image will not get loaded */

export const stateConstants = {
  STORE: 'STORE',
  OPERATION: 'OPERATION',
  AUTH: 'AUTH',
  PINSETUP: 'PINSETUP',
  BACKUPRESTORE: 'BACKUPRESTORE',
  CHANGEPASSWORD: 'CHANGEPASSWORD',
  APPCONTEXT: 'APPCONTEXT'
};

export class StoreConstants {
  static isInitialized = 'isInitialized';
  static loginAttempts = 'loginAttempts';
  static maxLoginAttempts = 6;
  /* In storage we use 'bewellapp_key' pattern for keys in key/value pairs */
  /* DataEncryptionStoreKey is special as it will not be hashed
  and the value will be encrypted with the user's password. When the user changes password
  only DataEncryptionStoreKey will have to be re-encrypted in store */
  static keyPrefix = 'bewellapp_';
  static keyDateFormat = 'MMYYYY';
  static monthsFromEpochDate = getMonthsFromEpochDate(StoreConstants.keyPrefix);
  static AllEncryptedStoreKeys = [...StoreConstants.monthsFromEpochDate];
  static DataEncryptionStoreKey = StoreConstants.keyPrefix + 'DATAENCRYPTIONKEY';
  static SETTINGS = StoreConstants.keyPrefix + 'SETTINGS';
}

function getMonthsFromEpochDate(keyPrefix: string) {
  /* data in store is partitioned by month so the keys can be potentially from
  when the app was first used to current year and month  ['012019', '022019', ... ] */
  const epochYear = 2019;
  const currentYear = new Date().getUTCFullYear();
  const currentMonth = new Date().getUTCMonth() + 1; /* getUTCMonth is zero based */
  const result = [];

  for (let year = epochYear; year <= currentYear; year++) {
    for (let month = 1; month <= 12; month++) {
      if (year === currentYear && month > currentMonth)
        break;
      result.push(keyPrefix + (month < 10 ? '0' : '') + month + '' + year); // 'MMYYYY' format
    };
  }
  return result;
}

export const settingsConstants = {
  language: 'language',
  theme: 'theme',
  version: 'version',
  hideNoteText: 'hideNoteText'
};

export enum ErrorCode {
  Encrypt1 = 'E1001',
  Encrypt2 = 'E1002',
  Encrypt3 = 'E1003',
  Encrypt4 = 'E1004',
  Decrypt1 = 'D1001',
  Decrypt2 = 'D1002',
  Decrypt3 = 'D1003',
  Decrypt4 = 'D1004',
  Decrypt5 = 'D1005',
  Decrypt6 = 'D1006',
  Decrypt7 = 'D1007',
  Decrypt8 = 'D1008',
  Decrypt9 = 'D1009',
  Decrypt10 = 'D10010',
  MissingItemType1 = 'MIT1001',
  MissingKey1 = 'MK1001',
  MissingKey2 = 'MK1002',
  MissingKey3 = 'MK1003',
  MissingKey4 = 'MK1004',
  MissingKey5 = 'MK1005',
  MissingKey7 = 'MK1007',
  MissingKey8 = 'MK1008',
  MissingKey9 = 'MK1009',
  MissingKey10 = 'MK10010',
  MissingKey11 = 'MK10011',
  UnableToHashWithoutPassword = 'HH1001',
  Hash1 = 'H1001',
  Hash2 = 'H1002',
  Storage1 = 'S1001',
  Storage2 = 'S1002',
  Storage3 = 'S1003',
  Storage4 = 'S1004',
  Storage5 = 'S1005',
  Storage6 = 'S1006',
  Storage7 = 'S1007',
  Storage8 = 'S108',
  Storage9 = 'S109',
  Storage10 = 'S1010',
  Storage11 = 'S1011',
  Storage12 = 'S1012',
  Auth1 = 'A1001',
  Auth2 = 'A1002',
  Auth3 = 'A1003',
  Auth4 = 'A1004',
  Auth5 = 'A1005',
  Auth6 = 'A1006',
  Auth7 = 'A1007',
  Auth8 = 'A1008',
  Auth9 = 'A1009',
  Auth10 = 'A10010',
  Auth11 = 'A10011',
  Auth12 = 'A10012',
  Security1 = 'SE1001',
  Security2 = 'SE1002',
  Security3 = 'SE1003',
  Security4 = 'SE1004',
  Security5 = 'SE1005',
  Security6 = 'SE1006',
  Security7 = 'SE1007',
  Security8 = 'SE1008',
  Security9 = 'SE1009',
  Security10 = 'SE1010',
  Import1 = 'I1001',
  Import2 = 'I1002',
  Import3 = 'I1003',
  BackupRestore1 = 'BR1001',
  BackupRestore2 = 'BR1002',
  BackupRestore3 = 'BR1003',
  BackupRestore4 = 'BR1004',
  File1 = 'F1001',
  File2 = 'F1002',
  File3 = 'F1003',
  File4 = 'F1004'
};
/**
 * @description Chooses an image from assets/images/headers directory
 * based on a random number and image filename. This is just for fun
 * instead of always showing the same header image. The filenames should
 * be 'header1.png', 'header2.png' etc format
 */
function getRandomHeaderImage() {
  const randomNumberForHeader = Math.floor(Math.random() * 11);  // returns a random integer from 0 to 10
  let headerImage;
  switch (randomNumberForHeader) {
    case 0:
      headerImage = require('../assets/images/headers/header1.png');
      break;
    case 1:
      headerImage = require('../assets/images/headers/header1.png');
      break;
    case 2:
      headerImage = require('../assets/images/headers/header2.png');
      break;
    case 3:
      headerImage = require('../assets/images/headers/header3.png');
      break;
    case 4:
      headerImage = require('../assets/images/headers/header4.png');
      break;
    case 5:
      headerImage = require('../assets/images/headers/header5.png');
      break;
    case 6:
      headerImage = require('../assets/images/headers/header6.png');
      break;
    case 7:
      headerImage = require('../assets/images/headers/header7.png');
      break;
    case 8:
      headerImage = require('../assets/images/headers/header8.png');
      break;
    case 9:
      headerImage = require('../assets/images/headers/header9.png');
      break;
    case 10:
      headerImage = require('../assets/images/headers/header10.png');
      break;
    default:
      headerImage = require('../assets/images/headers/header1.png');
      break;
  }
  return headerImage;
}