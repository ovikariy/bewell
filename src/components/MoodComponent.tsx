import React from 'react';
import { AppContext } from '../modules/appContext';
import { WidgetBase, WidgetComponentPropsBase, WidgetConfig } from '../modules/widgetFactory';
import { Text, View } from 'react-native';
import { friendlyTime } from '../modules/utils';
import { CustomIconType, CustomIconRating, CustomIconRatingItem } from './CustomIconRating';
import * as Animatable from 'react-native-animatable';
import { sizes } from '../assets/styles/style';

export interface MoodComponentWidgetType extends WidgetBase {
  rating?: number;
}

export interface MoodComponentProps extends WidgetComponentPropsBase {
  value: MoodComponentWidgetType;
  onChange: (newValue: MoodComponentWidgetType) => void
}

export const MoodComponent = (props: MoodComponentProps) => {
  const ratings = !props.config.icons ? <View /> : props.config.icons.map((icon: CustomIconType, index: number) => {
    return (
      <CustomIconRatingItem key={index} id={index} value={icon}
        selected={props.value && props.value.rating === index}
        onPress={(rating) => {
          if (!Number.isInteger(rating))
            return; //nothing to do since rating wasn't selected
          if (props.onChange)
            props.onChange({ ...props.value, rating });
        }} />
    );
  });
  return (
    <Animatable.View animation="fadeIn" duration={500}>
      <CustomIconRating>{ratings}</CustomIconRating>
    </Animatable.View>
  );
};

interface MoodHistoryComponentProps {
  item: MoodComponentWidgetType;
  isSelectedItem: boolean;
  config: WidgetConfig;
}
export const MoodHistoryComponent = (props: MoodHistoryComponentProps) => {
  const context = React.useContext(AppContext);
  const styles = context.styles;

  /* config is basically this object i.e config[itemType]  */
  /* custom render item to show mood icon in the row */
  const moodRatingIcons = props.config.icons;
  if (!moodRatingIcons || props.item.rating === undefined)
    return <View />;
  const ratingIcon = moodRatingIcons[props.item.rating];
  return (
    <View style={{ margin: sizes[10] }}>
      <Text style={props.isSelectedItem ? [styles.bodyText, styles.highlightColor] : styles.bodyText}>
        {friendlyTime(props.item.date)}</Text>
      <CustomIconRatingItem id={props.item.rating || -1} value={ratingIcon} textColor={props.isSelectedItem ? styles.highlightColor.color : null} />
    </View>
  );
};

export const MoodCalendarComponent = (props: MoodHistoryComponentProps) => {
  const context = React.useContext(AppContext);
  const styles = context.styles;

  /* config is basically this object i.e config[itemType]  */
  /* custom render item to show small mood icon in the calendar */
  const moodRatingIcons = props.config.icons;
  if (!moodRatingIcons || props.item.rating === undefined)
    return <View />;
  const ratingIcon = moodRatingIcons[props.item.rating];
  return <CustomIconRatingItem id={props.item.rating || -1} value={ratingIcon} hideText={true}
    iconStyle={styles.ratingIconSmallStyle}
    containerStyle={styles.ratingIconSmallContainer} />;
};