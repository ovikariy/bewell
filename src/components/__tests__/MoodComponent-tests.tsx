import 'react-native';
import * as mocks from '../../__mocks__/misc'; /** special mocks */
import React from 'react';
import renderer from 'react-test-renderer';
import { MoodComponent, MoodComponentWidgetType } from '../MoodComponent';
import { ItemTypes } from '../../modules/constants';
import { CreateWidgetFactory } from '../../modules/widgetFactory';
import { defaultAppContext } from '../../modules/appContext';

it('MoodComponent renders correctly', () => {
  const mood: MoodComponentWidgetType = {
    date: mocks.StableDate.toUTCString(),
    dateCreated: mocks.StableDate.toUTCString(),
    id: 'mood1',
    type: ItemTypes.MOOD,
  };
  const widgetFactory = CreateWidgetFactory(defaultAppContext);

  expect(renderer.create(
    <MoodComponent value={mood} onChange={() => { }} config={widgetFactory[ItemTypes.MOOD].config} selectedDate={mocks.StableDate} />
  )).toMatchSnapshot();

  mood.rating = 1;
  expect(renderer.create(
    <MoodComponent value={mood} onChange={() => { }} config={widgetFactory[ItemTypes.MOOD].config} selectedDate={mocks.StableDate}
      isSelected={true} key={mood.id} onSelected={() => { }} />
  )).toMatchSnapshot();
});
