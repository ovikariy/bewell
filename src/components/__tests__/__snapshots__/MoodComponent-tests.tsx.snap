// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodComponent renders correctly 1`] = `
<View
  forwardedRef={[Function]}
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "textAlign": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "transparent",
              "borderRadius": 100,
              "borderStyle": "dotted",
              "borderWidth": 3,
              "padding": 2,
            },
            "",
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ff9a55",
              "borderRadius": 100,
              "height": 54,
              "justifyContent": "center",
              "width": 54,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "color": "#87BCBF",
            "fontFamily": "Helvetica",
            "fontSize": 17,
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Happy
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "textAlign": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "transparent",
              "borderRadius": 100,
              "borderStyle": "dotted",
              "borderWidth": 3,
              "padding": 2,
            },
            "",
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#009898",
              "borderRadius": 100,
              "height": 54,
              "justifyContent": "center",
              "width": 54,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "color": "#87BCBF",
            "fontFamily": "Helvetica",
            "fontSize": 17,
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        So-so
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "textAlign": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "transparent",
              "borderRadius": 100,
              "borderStyle": "dotted",
              "borderWidth": 3,
              "padding": 2,
            },
            "",
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#517fa4",
              "borderRadius": 100,
              "height": 54,
              "justifyContent": "center",
              "width": 54,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "color": "#87BCBF",
            "fontFamily": "Helvetica",
            "fontSize": 17,
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Could be better
      </Text>
    </View>
  </View>
</View>
`;

exports[`MoodComponent renders correctly 2`] = `
<View
  forwardedRef={[Function]}
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "textAlign": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "transparent",
              "borderRadius": 100,
              "borderStyle": "dotted",
              "borderWidth": 3,
              "padding": 2,
            },
            "",
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ff9a55",
              "borderRadius": 100,
              "height": 54,
              "justifyContent": "center",
              "width": 54,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "color": "#87BCBF",
            "fontFamily": "Helvetica",
            "fontSize": 17,
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Happy
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "textAlign": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "transparent",
              "borderRadius": 100,
              "borderStyle": "dotted",
              "borderWidth": 3,
              "padding": 2,
            },
            Object {
              "borderColor": "#FFFFFF",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#009898",
              "borderRadius": 100,
              "height": 54,
              "justifyContent": "center",
              "width": 54,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "color": "#87BCBF",
            "fontFamily": "Helvetica",
            "fontSize": 17,
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        So-so
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "textAlign": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "transparent",
              "borderRadius": 100,
              "borderStyle": "dotted",
              "borderWidth": 3,
              "padding": 2,
            },
            "",
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#517fa4",
              "borderRadius": 100,
              "height": 54,
              "justifyContent": "center",
              "width": 54,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "color": "#87BCBF",
            "fontFamily": "Helvetica",
            "fontSize": 17,
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Could be better
      </Text>
    </View>
  </View>
</View>
`;
